torchbox/torchbox.com
# Historical Work Page | BE

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1370033202)

### Description of Changes Made

This refactors the existing Work Page model to pave way for the introduction of a new Work Page model based on the new designs.

**Key Changes**

- Rename `WorkPage` to `HistoricalWorkPage`
- Create `AbstractPageScreenShot` and `AbstractPageAuthor`, so that, upon subclassing, we can create a `page` `ParentalKey` to whichever `Page` model we choose. In this PR, we do this for the `HistoricalWorkPage` above. We'll be able to do this for the new `WorkPage`
- Remove some unused code (mainly tags on work pages)
- update references to `WorkPage` so that they now refer to `HistoricalWorkPage`
- Ensure that  `HistoricalWorkPage` cannot be created in Wagtail Admin (we want people to create the new `WorkPage`s instead)

### How to Test

In Wagtail Admin, go to Pages > Torchbox > Work: http://127.0.0.1:8000/admin/pages/6/. 

- You should see the old Work Pages there, and they should be called Historical Pages
- You should be able to edit the existing pages
- You should not be able to create a Historical Work Page

### Screenshots

<details>
  <summary>Expand to see more</summary>

![image](https://github.com/torchbox/torchbox.com/assets/7713776/80fa7e6b-1117-4b88-a2a0-7fa5a1926b82)


</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
